packages/relayer/indexer/filter_then_subscribe.go

```diff
@@ -107,7 +112,12 @@ func (svc *Service) FilterThenSubscribe(ctx context.Context, mode relayer.Mode)
 	}
 
 	if svc.processingBlock.Height < latestBlock.Number.Uint64() {
-		return svc.FilterThenSubscribe(ctx, relayer.SyncMode)
+		return svc.FilterThenSubscribe(ctx, relayer.SyncMode, watchMode)
+	}
+
+	// we are caught up and specified not to subscribe, we can return now
+	if watchMode == relayer.WatchMode(relayer.FilterWatchMode) {
```

Comment:
```suggestion
	if watchMode == relayer.FilterWatchMode {
```
unnecessary type conversion?